docs/source/amp.rst

```diff
@@ -0,0 +1,48 @@
+.. role:: hidden
+    :class: hidden-section
+
+Automatic Mixed Precision package - torch.cuda.amp
+==================================================
+
+.. automodule:: torch.cuda.amp
+.. currentmodule:: torch.cuda.amp
+
+``torch.cuda.amp`` provides convenience methods for "mixed precision:"  using ``torch.float16`` for some operations
+and ``torch.float32`` for others.  TODO discuss how we want to summarize this...right balance of technical and quick.
+
+The two ingredients of the mixed precision recipe are
+
+* `Autocasting`_, to ensure each operation runs in its optimal precision, and
+* `Gradient Scaling`_, to mitigate FP16 gradient underflow during the backward pass.
+
+By default, you don't need to call ``.half()`` on your model(s) or data to use the routines below.
+In fact, you shouldn't.  Model weights should remain FP32.
+
+Turning on mixed precision should not require retuning any hyperparameters, as long as the conventions shown in the
+:ref:`Automatic Mixed Precision Examples<amp-examples>` are obeyed.
+
+.. contents:: :local:
+
+Autocasting
+^^^^^^^^^^^
+
+Under construction...
+
+Gradient Scaling
+^^^^^^^^^^^^^^^^
+
+For additional guidance, please review whether any of the
+:ref:`Gradient Scaling Examples<gradient-scaling-examples>` match your use case.
+
+.. autoclass:: AmpScaler
+    :members:
+
+Custom Optimizer Guide
```

Comment:
"Writing an AMP Optimizer"

Reply:
Scaling is the only thing that requires customized treatment though.

Reply:
Your title is better.

Reply:
Thanks :)  I'll remember it for later.  Omitting custom optimizer stuff from the documents for now.